honeycombio/data_source_recipients.go

```diff
@@ -0,0 +1,107 @@
+package honeycombio
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	honeycombio "github.com/honeycombio/terraform-provider-honeycombio/client"
+	"github.com/honeycombio/terraform-provider-honeycombio/honeycombio/internal/hashcode"
+)
+
+func dataSourceHoneycombioRecipients() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceHoneycombioRecipientsRead,
+
+		Description: `
+'honeycombio_recipients' data source provides recipient IDs of recipients matching a set of criteria.
+`,
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
```

Comment:
As discussed, I think `type` should be required if there is a `detail_filter` block, as the filter sort of implicitly adds the type filter anyway. Best to make it explicit.